### ISSUE TITLE
V2.0 Idea: *User Library*

### ISSUE THREAD

**Issue by GoddessServant0** [ififfy/flipflip#48]
Ok, so this is kind of a long term project for me... but before Tumblr was being shut down, I was maintaining and developing a spreadsheet of tags and scripts to help me generate gooninator sessions by weighting the tags. I even had it automatically checking if tumblrs had been removed and blacklisting them. I loved it 😏

I implemented that gooninator import thing as a stopgap for now, but it only goes so far.

---

I'm proposing a "User Library" where you can import "Sources" for use in scenes. These could be local directories, tumblrs, subreddits, links to lists of files, etc. We can define types of sources and add more as we go along. Obviously, the player will need to be able to play these sources as well.

Users would have the ability to add tags to each source and somehow manage these tags. I'm not sure the absolute best way to manage these tags. I was thinking Users can add to a set of tags, and then -- when a source is selected -- a set of checkboxes appears, one for each tag. This, or an autocomplete textbox, would be best to reduce accidental duplication of tags.

The coup de gras will be a scene generator using these tags. A slider under each defined tag can define it's weight, along with other options like completely exclude a tag or solely include a tag. Then a new scene is generated automatically!

**Comment by ififfy** [comment]
I recommend keeping the basic data type of "sources" as simple URLs, and use regular expressions to figure out what they are. For example, if you add `https://goddessservant0.tumblr.com` as a source, it would know to use the Tumblr API to fetch the images, or if you use `https://reddit.com/r/flipflip`, it would load images from the FlipFlip subreddit. Then you don't need to sync up a bunch of source-related data types.

To sum up the basics of what you're saying: it sounds like you want to have one master list of sources, and add metadata to each source, rather than having to enter each source every time you make a scene. I agree it would be cool!
